node/external/blockAPI/internalBlock.go

```diff
@@ -82,13 +83,20 @@ func (ibp *internalBlockProcessor) GetInternalShardBlockByRound(format common.Ap
 	return ibp.convertShardBlockBytesByOutputFormat(format, blockBytes)
 }
 
-func (ibp *internalBlockProcessor) convertShardBlockBytesToInternalBlock(blockBytes []byte) (*block.Header, error) {
-	blockHeader := &block.Header{}
-	err := ibp.marshalizer.Unmarshal(blockHeader, blockBytes)
+func (ibp *internalBlockProcessor) convertShardBlockBytesToInternalBlock(blockBytes []byte) (interface{}, error) {
+	hdr, err := process.CreateShardHeader(ibp.marshalizer, blockBytes)
 	if err != nil {
 		return nil, err
 	}
+	blockHeaderV2, ok := hdr.(*block.HeaderV2)
```

Comment:
Please remove these casts and directly return the hdr.
Reason: If we add a new type of header (let's say v3) we need to remember that somewhere in the node's block api struct we have a duplicated logic only on some casts. 

The function will become:
```GO
func (ibp *internalBlockProcessor) convertShardBlockBytesToInternalBlock(blockBytes []byte) (interface{}, error) {
	return process.CreateShardHeader(ibp.marshalizer, blockBytes)
}
```

Reply:
since the header will be exposed to gin as json, it has to be concrete type, not interface, so in the end i have to cast it, or to have a separate function for this

Reply:
json can handle happily "interfaces". Remember that during execution, the will be concrete types flowing through here. Please do a test.

Reply:
thanks :+1: just checked and it works fine